File: frontend/src/models/transport/TransportRequest.ts
import type { RouteDto } from '@/models/route/Route'

export interface CreateTransportRequest {
    name: string
    customerId?: number

    driverId: number
    tractorId: number
    semiTrailerId?: number

    emptyTrip: boolean

    plannedStart: string
    plannedEnd: string

    originName: string
    originAddress: string
    originLat: number
    originLng: number

    destinationName: string
    destinationAddress: string
    destinationLat: number
    destinationLng: number
}

export interface CreateTransportFromRouteRequest {
    transport: CreateTransportRequest
    selectedRoute: RouteDto
}

export interface AssignTransport {
    title: string
    driverId: number
    customerId?: number
}

File: backend/src/main/java/com/optiroute/backend/dto/request/transport/TransportRequest.java
package com.optiroute.backend.dto.request.transport;

import java.time.OffsetDateTime;

public record TransportRequest(String name,

        Long driverId, Long tractorId, Long semiTrailerId, Long customerId,

        boolean emptyTrip,

        OffsetDateTime plannedStart, OffsetDateTime plannedEnd,

        String originName, String originAddress, double originLat, double originLng,

        String destinationName, String destinationAddress, double destinationLat, double destinationLng) {
}